test/python/test_square_octagon_color_code.py

```diff
@@ -0,0 +1,100 @@
+"""Test the square octagon color code. Created by Peter-Jan Derks."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import numpy as np
+import pytest
+
+from .test_utils import check_and_load_json
+
+if TYPE_CHECKING:
+    from _pytest.fixtures import FixtureRequest
+    from pytest_console_scripts import ScriptRunner
+
+from mqt.qecc.cc_decoder.square_octagon_color_code import SquareOctagonColorCode
+
+
+@pytest.fixture()
+def code(request: FixtureRequest) -> SquareOctagonColorCode:
+    """Color code on 4.8.8 lattice."""
+    return SquareOctagonColorCode(distance=request.param)
+
+
+@pytest.fixture()
+def d3_so_code() -> SquareOctagonColorCode:
+    """Distance of the hexagonal color code."""
+    return SquareOctagonColorCode(distance=3)
+
+
+@pytest.fixture()
+def p() -> float:
+    """Return error rate."""
+    return 0.1
+
+
+@pytest.fixture()
+def nr_sims() -> int:
+    """Return number of simulations."""
+    return 1
+
+
+@pytest.fixture()
+def results_dir() -> str:
+    """Return directory to store results."""
+    return "./results"
+
+
+@pytest.mark.parametrize("code", list(range(3, 23, 2)), indirect=True)
+def test_number_of_qubits(code: SquareOctagonColorCode) -> None:
```

Comment:
I think you do not really need the fixture on top, but you can directly make a parametrized test out of the Code objects